### PR TITLE
Avoid reseting the thread local interner at the beginning of `phase_1_parse_input`

### PR DESCRIPTION
**Pull request by jseyfried**
The thread local interner is used before `phase_1_parse_input` to create `InternedString`s, which currently wrap `Rc<String>`s. Once `InternedString` is refactored to be an interned string id (like `Name`), resetting will invalidate everything that was interned before `phase_1_parse_input`.

The resets were only useful for the `rusti` project, which can now use `driver::reset_thread_local_state`.

r? @nrc

### CODE REVIEW

**Comment by jseyfried** [comment]
cc @cswords @murarth
c.f. https://github.com/murarth/rusti/pull/83

**Comment by jseyfried** [comment]
I weakened the test that was causing problems on Mac and added a `FIXME` -- r? @nikomatsakis

File: src/test/compile-fail/lifetime-inference-give-expl-lifetime-param.rs [hunk 1]
```diff
@@ -49,7 +49,8 @@ struct Baz<'x> {
 
 impl<'a> Baz<'a> {
     fn baz2<'b>(&self, x: &isize) -> (&'b isize, &'b isize) {
-        //~^ HELP consider using an explicit lifetime parameter as shown: fn baz2<'b>(&self, x: &'a isize) -> (&'a isize, &'a isize)
+        //~^ HELP consider using an explicit lifetime parameter as shown: fn baz2<'b>(&self, x: &'
+        // FIXME #34856: The above suggestion is different on Linux and Mac.
```
Comment (nikomatsakis) [inline src/test/compile-fail/lifetime-inference-give-expl-lifetime-param.rs]: can you open an issue about the fact that this test is unrealiable and put the FIXME with that issue number?

Comment (jseyfried) [inline src/test/compile-fail/lifetime-inference-give-expl-lifetime-param.rs]: Will do

Comment (jseyfried) [inline src/test/compile-fail/lifetime-inference-give-expl-lifetime-param.rs]: Done.

**Comment by nikomatsakis** [comment]
@jseyfried r=me for that

**Pull request merged**